ttnn/cpp/ttnn/graph_processor.cpp

```diff
@@ -427,16 +427,47 @@ namespace ttnn {
             .params = {},
             .connections = {}
         });
-        hooks = std::make_shared<ProcessorHooks>();
-        tt::tt_metal::GraphTracker::instance().add_hook(hooks);
-        hooks->set_block(mode == RunMode::FAKE);
+
+        if (!tt::tt_metal::GraphTracker::instance().get_hook()) {
+            hooks = std::make_shared<ProcessorHooks>();
+            tt::tt_metal::GraphTracker::instance().add_hook(hooks);
```

Comment:
hook or hooks?

Reply:
ill fix it in the next pr.
Ill have more fixes.

Reply:
Any reason this can't be fixed in this PR that is introducing/renaming some of these things already?